feather/base/src/anvil/entity.rs

```diff
@@ -236,11 +238,19 @@ pub struct ItemNbt {
 
 impl ItemNbt {
     /// Create an `ItemStack` of the specified item and amount, setting any nbt present.
+    /// Panics if count is zero.
     pub fn item_stack(nbt: &Option<Self>, item: Item, count: u8) -> ItemStack {
-        ItemStack {
-            count: count as u32,
-            item,
-            damage: nbt.as_ref().map(|n| n.damage).flatten().map(|x| x as u32),
+        match nbt {
+            Some(ItemNbt {
+                damage: Some(damage),
+            }) => ItemStackBuilder::with_item(item)
+                .count(count as u32)
+                .damage((*damage).try_into().unwrap())
+                .into(),
+
+            Some(ItemNbt { damage: None }) | None => {
+                ItemStackBuilder::with_item(item).count(count as u32).into()
```

Comment:
If an `ItemStack` cannot have more than `u8` items, why is the builder using `u32`?

Reply:
`ItemStack.count` is defined as NonZeroU32. That is why it's being cast to u32 here.